src/components/multilineBox/MultilineBox.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import {createStructuredSelector} from 'reselect';
+import * as _ from 'underscore';
+import {IReactVaporState} from '../../ReactVapor';
+import {addStringList, addValueStringList, removeStringList, removeValueStringList} from '../../reusableState/customList/StringListActions';
+import {deepClone} from '../../utils/CloneUtils';
+import {IDispatch, ReduxConnect} from '../../utils/ReduxUtils';
+import {UUID} from '../../utils/UUID';
+import {IMultiSelectOwnProps} from '../select/MultiSelectConnected';
+import {MultilineBoxSelectors} from './MultilineBoxSelector';
+
+export interface IMultilineSingleBoxProps<T> {
+    id: string;
+    isLast: boolean;
+    props: T;
+}
+
+export interface IMultilineParentProps {
+    parentId: string;
+    removeBox: (id: string) => void;
+    addNewBox: () => void;
+}
+
+export interface IMultilineBoxOwnProps<T = any> {
+    id: string;
+    data: T[];
+    renderBody: (data: Array<IMultilineSingleBoxProps<T>>, parentProps: IMultilineParentProps) => React.ReactNode;
+    defaultProps: T;
+    renderWrapper?: (children: React.ReactNode, boxProps: IMultilineSingleBoxProps<T>, parentProps: IMultilineParentProps) => React.ReactNode;
+}
+
+export interface IMultilineBoxStateProps {
+    multilineBoxIds: string[];
+}
+
+export interface IMultilineBoxDispatchProps {
+    onMount: (defaultIds: string[]) => void;
+    onUnmount: () => void;
+    removeBox: (id: string) => void;
+    addNewBox: () => void;
+}
+
+export interface IMultilineBoxState<T> {
+    initialData: {[id: string]: T};
+}
+
+export interface IMultilineBoxProps<T = any> extends IMultilineBoxOwnProps<T>,
+    Partial<IMultilineBoxStateProps>,
+    Partial<IMultilineBoxDispatchProps> {}
+
+const makeMapStateToProps = () => {
+    const getStateProps = createStructuredSelector({
+        multilineBoxIds: MultilineBoxSelectors.getIds,
+    });
+
+    return (state: IReactVaporState, ownProps: IMultiSelectOwnProps): IMultilineBoxStateProps =>
+        getStateProps(state, {id: ownProps.id});
+};
+
+const mapDispatchToProps = (dispatch: IDispatch, ownProps: IMultilineBoxOwnProps): IMultilineBoxDispatchProps => ({
+    onMount: (defaultIDs: string[]) => dispatch(addStringList(ownProps.id, defaultIDs)),
+    onUnmount: () => dispatch(removeStringList(ownProps.id)),
+    removeBox: (id: string) => dispatch(removeValueStringList(ownProps.id, id)),
+    addNewBox: () => dispatch(addValueStringList(ownProps.id, UUID.generate())),
+});
+
+@ReduxConnect(makeMapStateToProps, mapDispatchToProps)
+export class MultilineBox<T> extends React.PureComponent<IMultilineBoxProps<T>, IMultilineBoxState<T>> {
+
+    constructor(props: IMultilineBoxProps<T>, state: IMultilineBoxState<T>) {
+        super(props, state);
+
+        this.state = {
+            initialData: this.getInitialDataMappedWithBoxIDs(),

```

Comment:
Using props to initialize the state is an anti-patter. You can read more about it here https://medium.com/@justintulk/react-anti-patterns-props-in-initial-state-28687846cc2e.

I suggest simply initializing initialData to an empty object:
In `constructor`:
```ts
this.state = {
    initialData: {},
};
```
Then in `componentDidMount`:
```ts
const defaultIds = this.getInitialDataMappedWithBoxIDs();
this.setState({initialData: defaultIds});
this.props.onMount(defaultIds);
```
And in `componentDidUpdate`:
```ts
if (prevProps && !_.isEqual(this.props.multilineBoxIds, prevProps.multilineBoxIds)) {
    this.setState({initialData: this.getInitialDataMappedWithBoxIDs()});
}
```